python/ray/serve/drivers.py

```diff
@@ -37,15 +37,34 @@ def __init__(
         self,
         dags: Union[RayServeDAGHandle, Dict[str, RayServeDAGHandle]],
         http_adapter: Optional[Union[str, Callable]] = None,
+        healthcheck_endpoint: Optional[str] = None,
```

Comment:
any reason this needs to be configurable? I'd prefer to just always make it `/healthz`

Reply:
The HTTP Proxy already has a `/healthz` endpoint, so requests to `/healthz` will get handled at the proxy without getting sent to the `DAGDriver`. I made it configurable, so it doesn't clash with any existing `route_prefixes` that users have.

Reply:
the HTTP proxy has `/-/healthz`, shouldn't conflict

Reply:
Nice, I can change it. Should the `DAGDriver` always expose a `/healthz` endpoint, or should the user enable it explicitly?

Reply:
I'd suggest we always expose it; actually, could we do this more generally for every ingress deployment? It can call `check_health` if the user provided it, else no-op.